projects/pt1/python/torch_mlir_e2e_test/test_suite/__init__.py

```diff
@@ -10,7 +10,8 @@
 from torch_mlir._version import torch_version_for_comparison, version
 
 COMMON_TORCH_MLIR_LOWERING_XFAILS = {
-    "NativeGroupNormModule_basic",
+    # "GroupNormModule_basic" failed because some list operator are not canonicalized.
+    "GroupNormModule_basic",
```

Comment:
I thought this was failing because of the use of an `torch.nn` module for the op. What error are you getting?

Reply:
Yes, it will generate some code like:
```
%0 = torch.vtensor.literal(dense<0.000000e+00> : tensor<4xf32>) : !torch.vtensor<[4],f32>
    %1 = torch.vtensor.literal(dense<1.000000e+00> : tensor<4xf32>) : !torch.vtensor<[4],f32>
    %float1.000000e-05 = torch.constant.float 1.000000e-05
    %str = torch.constant.str "Expected more than 1 value per channel when training, got input size {}"
    %str_0 = torch.constant.str "builtins.ValueError"
    %2 = torch.prim.ListConstruct %int4, %int2 : (!torch.int, !torch.int) -> !torch.list<int>
    %3 = torch.prim.ListConstruct %int6, %int7 : (!torch.int, !torch.int) -> !torch.list<int>
    %4 = torch.aten.list.t %3 : !torch.list<int> -> !torch.list<int>
    %5 = torch.aten.add.t %2, %4 : !torch.list<int>, !torch.list<int> -> !torch.list<int>
    %6 = torch.aten.__getitem__.t %5, %int0 : !torch.list<int>, !torch.int -> !torch.int
    %7 = torch.aten.len.t %5 : !torch.list<int> -> !torch.int
    %8 = torch.aten.sub.int %7, %int2 : !torch.int, !torch.int -> !torch.int
    %9 = torch.prim.Loop %8, %true, init(%6) {
    ^bb0(%arg1: !torch.int, %arg2: !torch.int):
      %37 = torch.aten.add.int %arg1, %int2 : !torch.int, !torch.int -> !torch.int
      %38 = torch.aten.__getitem__.t %5, %37 : !torch.list<int>, !torch.int -> !torch.int
      %39 = torch.aten.mul.int %arg2, %38 : !torch.int, !torch.int -> !torch.int
      torch.prim.Loop.condition %true, iter(%39 : !torch.int)
    } : (!torch.int, !torch.bool, !torch.int) -> !torch.int
    %10 = torch.aten.eq.int %9, %int1 : !torch.int, !torch.int -> !torch.bool
    torch.prim.If %10 -> () {
      %37 = torch.aten.format(%str, %5) : !torch.str, !torch.list<int> -> !torch.str
      torch.prim.RaiseException %37, %str_0 : !torch.str, !torch.str
      torch.prim.If.yield
    } else {
      torch.prim.If.yield
    }
```

If we want to resolve it, maybe we should add some canonicalize functions for `ListType`(but I think it's a little hard to resolve reference problem).  
So I agree to your idea. It's **no need** to add this testcase. It's a problem which would not appear in torchscript or which should be resolved by dynamo/fx.